src/main/webapp/app/shared/profile-toggle/profile-toggle.service.ts

```diff
@@ -0,0 +1,198 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable, OperatorFunction } from 'rxjs';
+import { JhiWebsocketService } from 'app/core/websocket/websocket.service';
+import { distinctUntilChanged, filter, map, pairwise, tap } from 'rxjs/operators';
+import { ProfileInfo } from 'app/shared/layouts/profiles/profile-info.model';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { ActivationStart, NavigationEnd, Router } from '@angular/router';
+import { AlertService, AlertType } from 'app/core/util/alert.service';
+
+/**
+ * ProfileToggle, currently only supports LECTURE
+ * @readonly
+ * @enum {string}
+ */
+export enum ProfileToggle {
+    LECTURE = 'lecture',
+    DECOUPLING = 'decoupling',
+}
+export type ActiveProfileToggles = Array<ProfileToggle>;
+
+@Injectable({ providedIn: 'root' })
+export class ProfileToggleService {
+    private infoUrl = 'management/info';
+
+    private readonly topic = `/topic/management/profile-toggles`;
+    private subject: BehaviorSubject<ActiveProfileToggles | undefined> = new BehaviorSubject<ActiveProfileToggles | undefined>(undefined);
+    private subscriptionInitialized = false;
+
+    private profileForCurrentRoute: ProfileToggle | undefined = undefined;
```

Comment:
```suggestion
    private profileForCurrentRoute?: ProfileToggle;
```
nitpick(?): should maybe work without the explicit `undefined`.